File: C1.UWP.FlexReport/CS/FlexReportExplorer/Data/CustomStyleSelector.cs
using C1.Xaml;
using Windows.UI.Xaml;
using Windows.UI.Xaml.Data;

namespace C1FlexReportExplorer
{
    public class CustomStyleSelector : C1StyleSelector
    {
        protected override Style SelectStyleCore(object item, DependencyObject container)
        {
            var tvi = (C1TreeViewItem)container;
            if (tvi.Header is Category)
            {
                tvi.SetBinding(C1TreeViewItem.IsExpandedProperty, new Binding()
                {
                    Source = tvi.Header,
                    Path = new PropertyPath("IsExpanded"),
                    Mode = BindingMode.TwoWay
                });
                return (Style)Resources["ExpandedStyle"];
            }
            if (tvi.Header is Report)
            {
                tvi.SetBinding(C1TreeViewItem.IsSelectedProperty, new Binding()
                {
                    Source = tvi.Header,
                    Path = new PropertyPath("IsSelected"),
                    Mode = BindingMode.TwoWay
                });
            }
            return null;
        }
    }
}

File: C1.UWP.FlexReport/VB/FlexReportExplorer/Data/CustomStyleSelector.vb
Imports C1.Xaml

Public Class CustomStyleSelector
    Inherits C1StyleSelector

    Protected Overrides Function SelectStyleCore(item As Object, container As DependencyObject) As Style
        Dim tvi = DirectCast(container, C1TreeViewItem)
        If TypeOf tvi.Header Is Category Then
            tvi.SetBinding(C1TreeViewItem.IsExpandedProperty, New Binding() With {
                .Source = tvi.Header,
                .Path = New PropertyPath("IsExpanded"),
                .Mode = BindingMode.TwoWay
            })
            Return DirectCast(Resources("ExpandedStyle"), Style)
        End If
        If TypeOf tvi.Header Is Report Then
            tvi.SetBinding(C1TreeViewItem.IsSelectedProperty, New Binding() With {
                .Source = tvi.Header,
                .Path = New PropertyPath("IsSelected"),
                .Mode = BindingMode.TwoWay
            })
        End If
        Return Nothing
    End Function

End Class
